### PR TITLE
Adding Pica8 support

### PR DESCRIPTION
**Pull request by packetshooter**
First attempt at contributing to an open source project by adding Pica8 support to Netmiko.

### CODE REVIEW

**Comment by packetshooter** [comment]
ran into issues while executing the tests, looking for some feedback to help me along. The test script has an error that it reports it can't exit config mode yet manually I am able to enter config mode and exit without issues:



(> is user mode while # is config mode)
Manual:
>>> device.find_prompt()
'netsetup@XorPlus>'
>>> device.config_mode()
'\rnetsetup@XorPlus> configure\r\r\n'
>>> device.find_prompt()
'netsetup@XorPlus#'
>>> device.exit_config_mode()
'\nnetsetup@XorPlus# exit configuration-mode\nnetsetup@XorPlus> '
>>> device.find_prompt()
'netsetup@XorPlus>'



Testing:
python3 -m pytest test_netmiko_config.py --test_device pica8
============================================================================ test session starts =============================================================================
platform darwin -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: /Users/ericvilleneuve/code/netmiko/netmiko
collected 6 items                                                                                                                                                            

test_netmiko_config.py ...F..                                                                                                                                          [100%]

================================================================================== FAILURES ==================================================================================
______________________________________________________________________________ test_command_set ______________________________________________________________________________

net_connect = <netmiko.pica8.pica8.pica8SSH object at 0x10d8fad68>
commands = {'basic': 'show interface brief', 'commit_verification': 'commit check', 'config': ['set system hostname test-sw', 'set system time-zone America/Edmonton'], 'config_verification': 'show | display set | match >', ...}
expected_responses = {'base_prompt': 'netsetup@XorPlus', 'commit_comment': 'Commit OK.', 'enable_prompt': 'netsetup@XorPlus#', 'interface_ip': '10.96.7.155', ...}

    def test_command_set(net_connect, commands, expected_responses):
        """Test sending configuration commands."""
        config_commands = commands["config"]
        support_commit = commands.get("support_commit")
        config_verify = commands["config_verification"]
    
        # Set to initial value and testing sending command as a string
>       net_connect.send_config_set(config_commands[0])

test_netmiko_config.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../netmiko/base_connection.py:1575: in send_config_set
    output += self.exit_config_mode()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <netmiko.pica8.pica8.pica8SSH object at 0x10d8fad68>, exit_config = 'exit configuration-mode'

    def exit_config_mode(self, exit_config='exit configuration-mode'):
        """Exit configuration mode."""
        output = ""
        if self.check_config_mode():
            output = self.send_command_timing(exit_config, strip_prompt=False, strip_command=False)
            if 'There are uncommitted changes?' in output:
                output += self.send_command_timing('exit discard', strip_prompt=False, strip_command=False)
            if self.check_config_mode():
>               raise ValueError("Failed to exit configuration mode")
E               ValueError: Failed to exit configuration mode

../netmiko/pica8/pica8.py:87: ValueError
============================================================== 1 failed, 5 passed in 36.05 seconds ===============================================================

File: PLATFORMS.md [hunk 1]
```diff
@@ -62,6 +62,7 @@
 - F5 TMSH
 - F5 Linux
 - Fortinet
+- Pica8
```
Comment (jdillard) [inline PLATFORMS.md]: That's not in alphabetical order like the rest

Comment (packetshooter) [reply PLATFORMS.md]: fixed the ordering in 58b45ad

**Comment by ktbyers** [comment]
@packetshooter It looks like you never really got this working? Let me know if you are still interested in working on it.

**Comment by packetshooter** [comment]
Hi Kirk, I am interested in working on it as we have moved to Pica8 as our OS for our switches. I’m back at work Monday and I can update the request and send you a note on slack.

Eric Villeneuve
Technical Analyst Team Leader
CALGARY CATHOLIC SCHOOL DISTRICT
1000 - 5 Avenue SW, Calgary, AB  T2P 4T9<x-apple-data-detectors://26/1>
Tel: 403-500-2408<tel:403-500-2408>
www.cssd.ab.ca<http://www.cssd.ab.ca/>

On Jan 1, 2020, at 10:28 PM, Kirk Byers <notifications@github.com> wrote:

﻿

@packetshooter<https://github.com/packetshooter> It looks like you never really got this working? Let me know if you are still interested in working on it.

**Comment by ktbyers** [comment]
Updated PR here:

https://github.com/ktbyers/netmiko/pull/1567

**Pull request closed without merging**